tap_facebook/streams/ad_insights.py

```
@@ -135,6 +138,36 @@ def _get_selected_columns(self) -> list[str]:
             columns = list(self.schema["properties"])
         return columns
 
+    def _get_earliest_record_date(self, account_id: str, sync_end_date: pendulum.Date) -> pendulum.Date | None:
+        """
+        Make a single Insights API call using sort to determine the oldest date with data.
+        Returns None if no data exists.
+        """
+        params = {
+            "level": self._report_definition["level"],
+            "fields": "date_start",
+            "sort": "date_start_ascending",
+            "time_range": {
+                "since": pendulum.parse(self.config["start_date"]).format("YYYY-MM-DD"),
+                "until": sync_end_date.format("YYYY-MM-DD"),
+            },
+        }
+        api: FacebookAdsApi = FacebookAdsApi.get_default_api()
+        url = f"https://graph.facebook.com/{self.config['api_version']}/act_{account_id}/insights"
+        try:
+            response = api.call("GET", url, params=params)
+            data = response.json().get("data", [])
+            if data:
+                earliest = pendulum.parse(data[0]["date_start"]).date()
+                self.logger.info(f"Earliest record found: {earliest.to_date_string()}")
+                return earliest
+            else:
+                self.logger.info("No data found for the specified date range")
+                return None
+        except Exception as e:
+            self.logger.error(f"Error fetching earliest record date: {e}")
+            return None
```

Why are we not throwing the error here?

If they have no data Facebook returns `{'data': []}`, which we handle correctly above

---

If there is a particular error we are looking for then we should do a hard check on the error message -- Otherwise we want to raise the error normally